mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -2845,6 +2845,20 @@ namespace {
 struct LLVMInlinerInterface : public DialectInlinerInterface {
   using DialectInlinerInterface::DialectInlinerInterface;
 
+  bool isLegalToInline(Operation *call, Operation *callable,
+                       bool wouldBeCloned) const final {
```

Comment:
Does `wouldBeCloned == false` imply that the original function will be removed? If so, we should maybe consider that as well.

Reply:
I now just return `false` is `wouldBeCloned == false`, since this doesn't seem safe in general and never occurs in practice. WDYT?

Reply:
As this is the safe approach, it seems good. If there is ever a need to change this, we can do so at that point in time.